From 9733fe32442ba6877dc502d89e90c6e9a08028d0 Mon Sep 17 00:00:00 2001
From: Jianqun Xu <jay.xu@rock-chips.com>
Date: Wed, 10 Aug 2016 11:11:29 +0800
Subject: [PATCH] ARM64: dts: rk3399: fix pwm id for vdd_center

RK3399 EVB1 and EVB2 use pwm3 for vdd_center, but EVB3 use pwm2.
This patch moved the vdd_center node to each board dtsi file.

Change-Id: I2b46b06b622c30ab65f26663a3628e73733472ad
Signed-off-by: Jianqun Xu <jay.xu@rock-chips.com>
---
 .../dts/rockchip/rk3399-evb-rev1-android.dts   | 10 ----------
 .../boot/dts/rockchip/rk3399-evb-rev1.dtsi     | 18 ++++++++++++++++++
 .../dts/rockchip/rk3399-evb-rev2-android.dts   |  5 -----
 .../boot/dts/rockchip/rk3399-evb-rev2.dtsi     | 16 ++++++++++++++++
 .../dts/rockchip/rk3399-evb-rev3-android.dts   |  5 -----
 .../boot/dts/rockchip/rk3399-evb-rev3.dtsi     | 16 ++++++++++++++++
 arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi   | 14 --------------
 7 files changed, 50 insertions(+), 34 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev1-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev1-android.dts
index 7226fa16ee2b..105e7a44aeae 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev1-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev1-android.dts
@@ -50,16 +50,6 @@
 	compatible = "rockchip,android", "rockchip,rk3399-evb-rev1", "rockchip,rk3399";
 };
 
-&vdd_log {
-	rockchip,pwm_id= <2>;
-	rockchip,pwm_voltage = <900000>;
-};
-
-&vdd_center {
-	rockchip,pwm_id= <3>;
-	rockchip,pwm_voltage = <900000>;
-};
-
 &rk_screen {
 	#include <dt-bindings/display/screen-timing/lcd-tv080wum-nl0-mipi.dtsi>
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev1.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev1.dtsi
index dc8653b93070..249d4d7ac811 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev1.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev1.dtsi
@@ -47,6 +47,8 @@
 
 	vdd_log: vdd-log {
 		compatible = "pwm-regulator";
+		rockchip,pwm_id = <2>;
+		rockchip,pwm_voltage = <900000>;
 		pwms = <&pwm2 0 25000 0>;
 		regulator-name = "vdd_log";
 		regulator-min-microvolt = <800000>;
@@ -54,6 +56,18 @@
 		regulator-always-on;
 		regulator-boot-on;
 	};
+
+	vdd_center: vdd-center {
+		compatible = "pwm-regulator";
+		rockchip,pwm_id = <3>;
+		rockchip,pwm_voltage = <900000>;
+		pwms = <&pwm3 0 25000 0>;
+		regulator-name = "vdd_center";
+		regulator-min-microvolt = <800000>;
+		regulator-max-microvolt = <1400000>;
+		regulator-always-on;
+		regulator-boot-on;
+	};
 };
 
 &cpu_l0 {
@@ -318,6 +332,10 @@
 	status = "okay";
 };
 
+&pwm3 {
+	status = "okay";
+};
+
 &u2phy0_otg {
 	rockchip,utmi-avalid;
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev2-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev2-android.dts
index 1f9e58c1423d..cf68431bb10a 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev2-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev2-android.dts
@@ -50,11 +50,6 @@
 	compatible = "rockchip,android", "rockchip,rk3399-evb-rev2", "rockchip,rk3399";
 };
 
-&vdd_center {
-	rockchip,pwm_id= <3>;
-	rockchip,pwm_voltage = <900000>;
-};
-
 &rk_screen {
 	#include <dt-bindings/display/screen-timing/lcd-tv080wum-nl0-mipi.dtsi>
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev2.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev2.dtsi
index 8fb8df2d27dc..941d75ac6f3e 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev2.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev2.dtsi
@@ -53,6 +53,18 @@
 		regulator-min-microvolt = <5000000>;
 		regulator-max-microvolt = <5000000>;
 	};
+
+	vdd_center: vdd-center {
+		compatible = "pwm-regulator";
+		rockchip,pwm_id = <3>;
+		rockchip,pwm_voltage = <900000>;
+		pwms = <&pwm3 0 25000 0>;
+		regulator-name = "vdd_center";
+		regulator-min-microvolt = <800000>;
+		regulator-max-microvolt = <1400000>;
+		regulator-always-on;
+		regulator-boot-on;
+	};
 };
 
 &cpu_l0 {
@@ -458,6 +470,10 @@
 	};
 };
 
+&pwm3 {
+	status = "okay";
+};
+
 &u2phy0_otg {
 	rockchip,utmi-avalid;
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev3-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev3-android.dts
index 6005f9bbf95c..8a09650ba372 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev3-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev3-android.dts
@@ -50,11 +50,6 @@
 	compatible = "rockchip,android", "rockchip,rk3399-evb-rev3", "rockchip,rk3399";
 };
 
-&vdd_center {
-	rockchip,pwm_id= <3>;
-	rockchip,pwm_voltage = <900000>;
-};
-
 &rk_screen {
 	#include <dt-bindings/display/screen-timing/lcd-tv080wum-nl0-mipi.dtsi>
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev3.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev3.dtsi
index 89bc3d491372..a151c739fe09 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev3.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev3.dtsi
@@ -53,6 +53,18 @@
 		regulator-min-microvolt = <5000000>;
 		regulator-max-microvolt = <5000000>;
 	};
+
+	vdd_center: vdd-center {
+		compatible = "pwm-regulator";
+		rockchip,pwm_id = <2>;
+		rockchip,pwm_voltage = <900000>;
+		pwms = <&pwm2 0 25000 0>;
+		regulator-name = "vdd_center";
+		regulator-min-microvolt = <800000>;
+		regulator-max-microvolt = <1400000>;
+		regulator-always-on;
+		regulator-boot-on;
+	};
 };
 
 &cluster0_opp {
@@ -472,3 +484,7 @@
 		status = "okay";
 	};
 };
+
+&pwm2 {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi
index 122f28cbc040..0b67e7a0d75f 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi
@@ -46,16 +46,6 @@
 / {
 	compatible = "rockchip,rk3399-evb", "rockchip,rk3399";
 
-	vdd_center: vdd-center {
-		compatible = "pwm-regulator";
-		pwms = <&pwm3 0 25000 0>;
-		regulator-name = "vdd_center";
-		regulator-min-microvolt = <800000>;
-		regulator-max-microvolt = <1400000>;
-		regulator-always-on;
-		regulator-boot-on;
-	};
-
 	vcc3v3_sys: vcc3v3-sys {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc3v3_sys";
@@ -467,10 +457,6 @@
 	status = "okay";
 };
 
-&pwm3 {
-	status = "okay";
-};
-
 &gmac {
 	phy-supply = <&vcc_phy>;
 	phy-mode = "rgmii";
-- 
2.35.3

